core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala

```diff
@@ -0,0 +1,214 @@
+                      stat: Stat) {
+      val zkUtils: ZkUtils = ctx.asInstanceOf[ZkUtils]
+      val zkHandle = zkUtils.zkConnection.getZookeeper
+        
+      Code.get(rc) match {
+          case Code.OK =>
+            info("Successfully set ACLs for %s".format(path))
+            counter.decrementAndGet()
+          case Code.CONNECTIONLOSS =>
+            Future {
+              zkHandle.setACL(path, ZkUtils.DefaultAcls(true), -1, SetACLCallback, ctx)
+            }(threadExecutionContext)
+          case Code.NONODE =>
+            warn("Node is gone, it could be have been legitimately deleted: %s".format(path))
+          case Code.SESSIONEXPIRED =>
+            // Starting a new session isn't really a problem, but it'd complicate
+            // the logic of the tool, so we quit and let the user re-run it.
+            error("ZooKeeper session expired while changing ACLs")
+            exception = throw ZkException.create(KeeperException.create(Code.get(rc)))
+          case _ =>
+            error("Unexpected return code: %d".format(rc))
+            exception = throw ZkException.create(KeeperException.create(Code.get(rc)))   
+      }
+    }
+  }
+  
+  def outputUsage(parser: OptionParser) {
+    CommandLineUtils.printUsageAndDie(parser, "ZooKeeper Migration Tool Help. This tool updates the ACLs of "
+                                                + "znodes as part of the process of setting up ZooKeeper "
+                                                + "authentication.")
+  }
+  
+  def run(args: Array[String]) {
+    var jaasFile = System.getProperty(JaasUtils.JAVA_LOGIN_CONFIG_PARAM)
+    val parser = new OptionParser()
+
+    val goOpt = parser.accepts("kafka.go", "Indicates whether to make this cluster secure or unsecure. The "
+        + " options are 'secure' and 'unsecure'").withRequiredArg().ofType(classOf[String])
+    val jaasFileOpt = parser.accepts("jaas.file", "JAAS Config file.").withOptionalArg().ofType(classOf[String])
+    val zkUrlOpt = parser.accepts("zookeeper.connect", "Sets the ZooKeeper connect string (ensemble). This parameter " +
+      "takes a comma-separated list of host:port pairs.").withRequiredArg().defaultsTo("localhost:2181").
+      ofType(classOf[String])
+    val zkSessionTimeoutOpt = parser.accepts("zookeeper.session.timeout", "Sets the ZooKeeper session timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val zkConnectionTimeoutOpt = parser.accepts("zookeeper.connection.timeout", "Sets the ZooKeeper connection timeout.").
+      withRequiredArg().defaultsTo("30000").ofType(classOf[java.lang.Integer])
+    val helpOpt = parser.accepts("help", "Print usage information.")
+
+    val options = parser.parse(args : _*)
+    if(options.has(helpOpt))
+      outputUsage(parser)
+
+    if ((jaasFile == null) && !options.has(jaasFileOpt)) {
+      error("No JAAS configuration file has been specified. Please make sure that you have set either "
```

Comment:
One question regarding the usage of the tool and the logging. Many of our command-line tools seem to use standard out and standard err directly instead of a logger. A logger can be more versatile, but we would ideally format it for console usage instead of the typical log format. But maybe this is not a command-line tool? What are your thoughts?